client/state/timezones/reducer.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	DESERIALIZE,
+	SERIALIZE,
+
+	TIMEZONES_RECEIVE,
+	TIMEZONES_REQUEST,
+	TIMEZONES_REQUEST_FAILURE,
+	TIMEZONES_REQUEST_SUCCESS,
+} from 'state/action-types';
+import {
+	rawOffsetsSchema,
+	timezonesSchema
+} from './schema';
+
+const offsets = createReducer( [], {
+	[ TIMEZONES_RECEIVE ]: ( state, { rawOffsets = [] } ) => rawOffsets.slice( 0 )
+}, rawOffsetsSchema );
+
+const timezones = createReducer( {}, {
+	[ TIMEZONES_RECEIVE ]: ( state, { timezonesByContinent } ) => ( timezonesByContinent )
+}, timezonesSchema );
+
+export const items = combineReducers( {
```

Comment:
I was thinking perhaps a structure like:

```json
{
	"labels": {
		"Africa/Abidjan": "Abidjan",
		"Africa/Accra": "Accra"
	},
	"continents": {
		"Africa": [
			"Africa/Abidjan",
			"Africa/Accra"
		]
	}
}
```

With selectors targeted toward the desired UI usage:

```js
function getTimezoneLabel( state, key ) {
	return state.timezones.labels[ key ];
}
```

```js
function getTimezonesByContinent( state, continent ) {
	return state.timezones.continents[ continent ];
}
```

```js
function getTimezoneLabelsByContinent( state, continent ) {
	const timezones = getTimezonesByContinent( state, continent );
	return fromPairs( map( timezones, ( key ) => [ key, getTimezone( state, key ) ] ) );
}
```

_(Perhaps merging the first two selectors into the last if we don't foresee them being used separately)_